src/main/java/io/citrine/jcc/search/dataset/result/DatasetMultiSearchResultElement.java

```diff
@@ -0,0 +1,23 @@
+package io.citrine.jcc.search.dataset.result;
+
+import io.citrine.jcc.search.core.result.BaseMultiSearchResultElement;
+
+/**
+ * Class to store the information in a single search result as a part of a dataset multi-query.
+ *
+ * @author Kyle Michel
+ */
+public class DatasetMultiSearchResultElement extends BaseMultiSearchResultElement<DatasetSearchResult> {
```

Comment:
The closest guidance I can find in effective java is "#26: favor generic types"; it doesn't say anything about APIs.